src/lib/Elf.cc

```diff
@@ -81,12 +81,86 @@ Elf::Elf(std::string path) {
   file.close();
 }
 
-Elf::~Elf() {
-  if (isValid_) {
-    delete[] processImage_;
+Elf::Elf(std::string path, std::vector<char>& imageVec) {
```

Comment:
I assume apart from the use of `imageVec`, the functionality/code is identical to the original ELF constructor?

Reply:
Yes.